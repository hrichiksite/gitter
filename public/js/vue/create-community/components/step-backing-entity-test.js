const mount = require('../../__test__/vuex-mount');
const { default: StepBackingEntity } = require('./step-backing-entity.vue');

import {
  CREATE_COMMUNITY_STEP_MAIN,
  CREATE_COMMUNITY_ENTITY_TYPE_TAB_ORGS_STATE,
  CREATE_COMMUNITY_ENTITY_TYPE_TAB_REPOS_STATE
} from '../constants';

const DEFAULT_PROPS = {
  orgName: 'orgs',
  orgList: [],
  repoName: 'repos',
  repoList: []
};

describe('StepBackingEntity', () => {
  it('orgs tab matches snapshot', () => {
    const { wrapper } = mount(StepBackingEntity, DEFAULT_PROPS, store => {
      store.state.createCommunity.entityTypeTabState = CREATE_COMMUNITY_ENTITY_TYPE_TAB_ORGS_STATE;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('repos tab matches snapshot', () => {
    const { wrapper } = mount(StepBackingEntity, DEFAULT_PROPS, store => {
      store.state.createCommunity.entityTypeTabState = CREATE_COMMUNITY_ENTITY_TYPE_TAB_REPOS_STATE;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('click tabs', () => {
    it('should switch to org tab when you click on it', () => {
      const { wrapper, stubbedActions } = mount(StepBackingEntity, DEFAULT_PROPS, store => {
        store.state.createCommunity.entityTypeTabState = CREATE_COMMUNITY_ENTITY_TYPE_TAB_REPOS_STATE;
      });
      wrapper.find({ ref: 'orgTabButton' }).trigger('click');

      expect(stubbedActions.createCommunity.setEntityTypeTabState).toHaveBeenCalledWith(
        expect.anything(),
        CREATE_COMMUNITY_ENTITY_TYPE_TAB_ORGS_STATE,
        undefined
      );
    });

    it('should switch to repo tab when you click on it', () => {
      const { wrapper, stubbedActions } = mount(StepBackingEntity, DEFAULT_PROPS, store => {
        store.state.createCommunity.entityTypeTabState = CREATE_COMMUNITY_ENTITY_TYPE_TAB_ORGS_STATE;
      });
      wrapper.find({ ref: 'repoTabButton' }).trigger('click');

      expect(stubbedActions.createCommunity.setEntityTypeTabState).toHaveBeenCalledWith(
        expect.anything(),
        CREATE_COMMUNITY_ENTITY_TYPE_TAB_REPOS_STATE,
        undefined
      );
    });
  });

  it('should go back to main view with back button clicked', () => {
    const { wrapper, stubbedActions } = mount(StepBackingEntity, DEFAULT_PROPS);
    wrapper.find({ ref: 'backButton' }).trigger('click');

    expect(stubbedActions.createCommunity.moveToStep).toHaveBeenCalledWith(
      expect.anything(),
      CREATE_COMMUNITY_STEP_MAIN,
      undefined
    );
  });
});
