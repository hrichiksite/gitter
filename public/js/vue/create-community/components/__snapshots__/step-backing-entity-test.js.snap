// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepBackingEntity orgs tab matches snapshot 1`] = `
<div>
  <h2
    class="secondary-community-heading"
  >
    
    Use any of your orgs or repos to create a community
  
  </h2>
   
  <ul
    class="tabbed-navigation-list scroller"
  >
    <li
      class="tabbed-navigation-list-item"
    >
      <button
        class="tabbed-navigation-button active"
      >
        
        orgs
      
      </button>
    </li>
     
    <li
      class="tabbed-navigation-list-item"
    >
      <button
        class="tabbed-navigation-button"
      >
        
        repos
      
      </button>
    </li>
  </ul>
   
  <entity-list-stub
    displayname="orgs"
    list=""
  />
   
  <!---->
    
  <button
    class="back-button secondary-button-clouds"
  >
    
    Back
  
  </button>
</div>
`;

exports[`StepBackingEntity repos tab matches snapshot 1`] = `
<div>
  <h2
    class="secondary-community-heading"
  >
    
    Use any of your orgs or repos to create a community
  
  </h2>
   
  <ul
    class="tabbed-navigation-list scroller"
  >
    <li
      class="tabbed-navigation-list-item"
    >
      <button
        class="tabbed-navigation-button"
      >
        
        orgs
      
      </button>
    </li>
     
    <li
      class="tabbed-navigation-list-item"
    >
      <button
        class="tabbed-navigation-button active"
      >
        
        repos
      
      </button>
    </li>
  </ul>
   
  <!---->
   
  <entity-list-stub
    displayname="repos"
    list=""
  />
    
  <button
    class="back-button secondary-button-clouds"
  >
    
    Back
  
  </button>
</div>
`;
